front/src/modules/timesStops/hooks/useOutputTableData.tsx

```diff
@@ -0,0 +1,102 @@
+  operationalPoints: OperationalPointWithTimeAndSpeed[],
+  selectedTrainSchedule: TrainScheduleResult,
+  startTime: string,
+  pathLength?: number
+) {
+  const scheduleByAt = keyBy(selectedTrainSchedule.schedule, 'at');
+  const suggestedOperationalPoints: SuggestedOP[] = useMemo(
+    () =>
+      pathLength
+        ? formatSuggestedOperationalPoints(
+            pathProperties.operationalPoints,
+            pathProperties.geometry,
+            pathLength
+          )
+        : [],
+    [pathProperties.operationalPoints, pathProperties.geometry]
+  );
+
+  const pathStepsWithOpPointIndices = selectedTrainSchedule.path.map((pathStep) => {
+    const correspondingOpPointIndex = suggestedOperationalPoints.findIndex(
+      (sugOpPoint) =>
+        'uic' in pathStep &&
+        sugOpPoint.uic === pathStep.uic &&
+        sugOpPoint.ch === pathStep.secondary_code
+    );
+    return {
+      ...pathStep,
+      correspondingOpPointIndex,
+    };
+  });
+  const trainSchedulePathByUic = keyBy(pathStepsWithOpPointIndices, (pathStep) => {
+    if ('uic' in pathStep) {
+      return `${pathStep.uic}-${pathStep.secondary_code}`;
+    }
+    if ('operational_point' in pathStep) {
+      return pathStep.operational_point;
+    }
+    return '';
+  });
+
+  return suggestedOperationalPoints.map((sugOpPoint, sugOpIndex) => {
+    const opPoint = operationalPoints[sugOpIndex];
+    const nextOpPoint = findNextScheduledPoint(
+      operationalPoints,
+      pathStepsWithOpPointIndices,
+      sugOpIndex
+    );
+    const pathKey = `${sugOpPoint?.uic}-${sugOpPoint?.ch}`;
+    const pathStep = trainSchedulePathByUic[pathKey];
+
+    if (pathKey in trainSchedulePathByUic) {
+      const pathId = 'id' in pathStep ? pathStep.id : '';
+      const schedule = scheduleByAt[pathId];
+      const arrival = schedule?.arrival ? ISO8601Duration2sec(schedule.arrival!) : null;
+      const stopFor = schedule?.stop_for ? ISO8601Duration2sec(schedule.stop_for!) : '';
+      const departure = arrival && stopFor ? secToHoursString(arrival + stopFor) : null;
+      const [theoreticalMargin, realMargin, diffMargins] = useComputedMargins(
+        simulatedTrain,
+        opPoint,
+        nextOpPoint,
+        selectedTrainSchedule,
+        pathId
+      );
+      return {
+        ...sugOpPoint,
+        arrival: arrival ? secToHoursString(datetime2sec(new Date(startTime)) + arrival, true) : '',
+        departure,
+        stopFor: String(stopFor), // I have some numbers in the input table state
+        onStopSignal: schedule?.on_stop_signal || '',
+        calculatedArrival: secToHoursString(opPoint?.time, true),
+        calculatedDeparture:
+          opPoint?.duration > 0 ? secToHoursString(opPoint?.time + opPoint?.duration, true) : '',
+        theoreticalMargin,
+        calculatedMargin: realMargin,
+        diffMargins,
+      } as SuggestedOP;
+    }
+
+    return { ...sugOpPoint, calculatedArrival: secToHoursString(opPoint?.time, true) };
```

Comment:
Remove the ? if you have added an early return
```suggestion
    return { ...sugOpPoint, calculatedArrival: secToHoursString(opPoint.time, true) };
```

Reply:
yes